DependencyInjection/Compiler/AnnotationConfigurationPass.php

```diff
@@ -55,14 +55,12 @@ public function process(ContainerBuilder $container)
             if (null === $metadata = $factory->getMetadataForClass($className)) {
                 continue;
             }
+
             if (null === $metadata->getOutsideClassMetadata()->id) {
                 continue;
             }
-            if ( ! $metadata->getOutsideClassMetadata()->isLoadedInEnvironment($container->getParameter('kernel.environment'))) {
-                continue;
-            }
 
-            foreach ($converter->convert($metadata) as $id => $definition) {
+            foreach ($converter->convert($metadata, $container->getParameter('kernel.environment')) as $id => $definition) {
```

Comment:
before the environment check could be done on class level, but since we have now multiple services per class we have to check in there